src/app/state/wallet/saga.test.ts

```diff
@@ -70,6 +71,40 @@ describe('Wallet Sagas', () => {
         .silentRun(50)
     })
 
+    it('Should open from ledger and select the first imported wallet as active', () => {
+      return expectSaga(rootWalletSaga)
+        .provide(providers)
+        .withState({
+          wallet: {
+            isOpen: true,
+            selectedWallet: 0,
+            wallets: [
+              {
+                address: 'oasis1qz0k5q8vjqvu4s4nwxyj406ylnflkc4vrcjghuwk',
+                id: 0,
+              },
+              {
+                address: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4',
+                id: 1,
```

Comment:
Does this need to be here to work around `walletId` global variable messing everything up?

Reply:
It's not related to global variable. Test covers another issue when user imports wallets and the first selected wallet exists in redux store. In this case we use existing id not a new one stored in global `walletId` variable. 

Steps to repro (selection modal): 
A - selected
B - selected  
C 
D
E - selected

- nav to import again 
A
B - selected  
C - selected  
D - selected  
E - selected